SConstruct

```diff
@@ -732,6 +738,23 @@ def should_prepare_targets(style, toolchain, variant):
         if should_prepare_target(t, style, toolchain, variant):
             return True
 
+def should_build_ninja(style, toolchain, variant):
```

Comment:
I'm the one who's going to have to port this to the new scons system, so I'd love some documentation of exactly what it's doing.